Week3/homework/5totalCost.js

```diff
@@ -0,0 +1,31 @@
+// **Exercise 5: Total cost is ...**
+//
+// You want to buy a couple of things from the supermarket to prepare for a party. After scanning all the items the cashier gives you the total price, but the machine a broken! Let's write her a function that does it for her instead!
+//
+// 1. Write a function called `calculateTotalPrice`
+//
+// - It takes 1 argument: an object that contains properties that only contain number values
+// - Add all the number values together
+// - Return a number: the total price of all items
+//
+// 2. Create an object named `cartForParty` with 5 properties. Each property should be a grocery item (like `beers` or `chips`) and hold a number value (like `1.75` or `0.99`)
+// 3. Call the function 1 time, giving it the object `cartForParty` as an argument
+
+function calculateTotalPrice(listObject){
+  let total = 0;
+  for (var item in listObject) {
+    total += listObject[item];
+  }
+  return total;
```

Comment:
👍 for logic but I would like to introduce to you a couple new methods here which you will learn about in JS2 or advanced JS.
- `reduce`(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)
- `Object.keys`(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)

Perhaps you can explore how to use these two methods to replace your current logic of calculating total price because using `for-in` to iterate over objects is an old way of doing in JavaScript. This is because this loop also iterates over all inherited properties of the object rather than focussing only on the object's own properties. It's an advanced topic which you will learn in JS3 but it is good to know before 😉

Reply:
Thanks @yash-kapila . I'll try that one as well. So, that is why it is recommended to check if it is a property of the object. Good to know.